synapse/rest/media/v1/thumbnailer.py

```diff
@@ -129,24 +118,62 @@ def crop(self, width: int, height: int, output_type: str) -> BytesIO:
         Returns:
             BytesIO: the bytes of the encoded image ready to be written to disk
         """
+
+        return self._process_animated_aware(
+            self.image, output_type, lambda i: self._crop(i, width, height)
+        )
+
+    def _process_animated_aware(
+        self, image: Image, output_type: str, transform: "Callable[[Image], Image]"
+    ) -> BytesIO:
+        if self.FORMATS[output_type] == "GIF" and getattr(image, "n_frames", 1) > 1:
+            first, *rest = (transform(i) for i in ImageSequence.Iterator(image))
+            return self._encode_image(first, output_type, append_images=rest)
+        else:
+            return self._encode_image(transform(image), output_type)
+
+    def _crop(self, image: Image, width: int, height: int) -> Image:
```

Comment:
to-private and addition of `image` to decouple it and make it able to be used as a generic image transformer